src/components/sidebar/components/sidebar-nav/sidebar-nav-menu-item/index.tsx

```diff
@@ -40,7 +47,18 @@ const SidebarNavSubmenu: React.FC<{
   item: MenuItem
 }> = ({ item }) => {
   const buttonRef = useRef<HTMLButtonElement>()
-  const [isOpen, setIsOpen] = useState(item.hasActiveChild)
+  const [isOpen, setIsOpen] = useState(
+    item.hasActiveChild || item.hasChildrenMatchingFilter
+  )
+
+  /**
+   * Without this effect, the menu items aren't re-rerendered to be open. Seems
+   * to be because the item prop sent to the component don't change. Might work
+   * if we pass the props needed instead of just the item object?
+   */
+  useEffect(() => {
+    setIsOpen(item.hasActiveChild || item.hasChildrenMatchingFilter)
+  }, [item.hasActiveChild, item.hasChildrenMatchingFilter])
```

Comment:
I think this would still be necessary, because this component also acts as a "controlled" component, where its open state could be changed at any point by its props, not just the initial props. What we've got here is the right approach!